apps/music/test/marionette/Player_test.js

```diff
@@ -185,4 +185,23 @@ marionette('Music player tests', function() {
     });
   });
 
+  suite('Player navigation', function() {
+    test('Check back button works', function() {
+      music.launch();
+      music.waitForFirstTile();
+      music.switchToSongsView();
+
+      var title = music.header.findElement('#title-text').text();
+
+      music.playFirstSong();
+
+      assert.notEqual(title, music.header.findElement('#title-text').text());
+
+      music.tapHeaderActionButton();
+
```

Comment:
Do you need to do any kind of wait here, or does the Music app change its title synchronously when the tap event is received?  If it waits for a CSS transition or a db query to finish, then the change might not happen immediately.